scm/GitManager.go

```diff
@@ -38,35 +35,144 @@ func (a GitManager) Purge(dir bugs.Directory) error {
 	cmd.Stdin = os.Stdin
 	cmd.Stdout = os.Stdout
 	cmd.Stderr = os.Stderr
-	err := cmd.Run()
+	return cmd.Run()
+}
 
-	if err != nil {
-		return err
-	}
-	return nil
+type issueStatus struct {
+	a, d, m bool // Added, Deleted, Modified
 }
+type issuesStatus map[string]issueStatus
+
+// Get list of created, updated, closed and closed-on-github issues.
+//
+// In general following rules to categorize issues are applied:
+// * closed if Description file is deleted (D);
+// * created if Description file is created (A) (TODO: handle issue renamings);
+// * closed issue will also close issue on GH when Autoclose is true (see Identifier example);
+// * updated if Description file is modified (M);
+// * updated if Description is unchanged but any other files are touched. (' '+x)
+//
+// eg output from `from git status --porcelain`, appendix mine
+// note that `git add -A issues` was invoked before
+//
+// D  issues/First-GH-issue/Description		issue closed (GH issues are also here)
+// D  issues/First-GH-issue/Identifier		maybe it is GH issue, maybe not
+// M  issues/issue--2/Description		desc updated
+// A  issues/issue--2/Status			new field added (status); considered as update unless Description is also created
+// D  issues/issue1/Description			issue closed
+// A  issues/issue3/Description			new issue, description field is mandatory
+func (a GitManager) currentStatus(dir bugs.Directory) (closedOnGitHub []string, _ issuesStatus) {
+	ghRegex := regexp.MustCompile("(?im)^-Github:(\\s*)(.*)(\\s*)$^")
+	closesGH := func(file string) (issue string, ok bool) {
+		if !a.Autoclose {
+			return "", false
+		}
+		if !strings.HasSuffix(file, "Identifier") {
+			return "", false
+		}
+		diff := exec.Command("git", "diff", "--staged", "--", file)
+		diffout, _ := diff.CombinedOutput()
+		matches := ghRegex.FindStringSubmatch(string(diffout))
+		if len(matches) > 2 {
+			return matches[2], true
+		}
+		return "", false
+	}
+	short := func(path string) string {
+		b := strings.Index(path, "/")
+		e := strings.LastIndex(path, "/")
+		if b+1 >= e {
+			return "???"
+		}
+		return path[b+1 : e]
+	}
 
-func (a GitManager) getDeletedIdentifiers(dir bugs.Directory) []string {
 	cmd := exec.Command("git", "status", "-z", "--porcelain", string(dir))
 	out, _ := cmd.CombinedOutput()
 	files := strings.Split(string(out), "\000")
-	retVal := []string{}
+
+	issues := issuesStatus{}
+	var ghClosed []string
 	for _, file := range files {
-		if file == "" {
+		if len(file) < 5 {
```

Comment:
There must be at least 3 chars for variable _path_, I put margin somehow higher (it could be even 8 I think, but it will be even more magic :P)